File: PersonRegistration.Server/Models/Person.cs
namespace PersonRegistration.Server.Models;

public class Person
{
    public int Id { get; set; }

    public required string Name { get; set; }
    public required string Cpf { get; set; }
    public required DateOnly BirthDate { get; set; }

    public string? Email { get; set; }
    public string? BirthPlace { get; set; }
    public string? Nationality { get; set; }
    public string? Address { get; set; }
    public BirthSex? BirthSex { get; set; }

    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}

File: personregistration.client/src/types/person.ts
export type BirthSex = "Masculino" | "Feminino" | "Prefiro não responder";

export interface Person {
  id: number;
  name: string;
  cpf: string;
  birthDate: string;
  birthSex: BirthSex | null;
  email: string | null;
  birthPlace: string | null;
  nationality: string | null;
  address: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface PersonRequest {
  name: string;
  cpf: string;
  birthDate: string | null;
  birthSex: BirthSex | null;
  email: string | null;
  birthPlace: string | null;
  nationality: string | null;
  address?: string | null;
}